patches/server/0877-Add-PlayerInsertLecternBookEvent.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Mariell Hoversholm <proximyst@proximyst.com>
+Date: Sat, 29 Aug 2020 21:55:50 +0200
+Subject: [PATCH] Add PlayerInsertLecternBookEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/LecternBlock.java b/src/main/java/net/minecraft/world/level/block/LecternBlock.java
+index 25ed6dfef2887612a02fcf8884dc8dac4fbd64ff..7d1544f51bed51e625440d197df953a0ed831dae 100644
+--- a/src/main/java/net/minecraft/world/level/block/LecternBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/LecternBlock.java
+@@ -134,6 +134,7 @@ public class LecternBlock extends BaseEntityBlock {
+         if (!(Boolean) state.getValue(LecternBlock.HAS_BOOK)) {
+             if (!world.isClientSide) {
+                 LecternBlock.placeBook(player, world, pos, state, stack);
++                return ((LecternBlockEntity) world.getBlockEntity(pos)).hasBook(); // Paper
```

Comment:
I would do an instance check on the result of world.getBlockEntity(pos) here as something else might cancel the block placement, or even in the PlayerInsertLecternBookEvent they change the block and remove the block entity. Like all over Mojang changes blocks, and then they check if the tile entity was created, they don't assume it was. Look at the first couple lines of placeBook